src/OrbitVulkanLayer/VulkanLayerController.h

```diff
@@ -704,15 +704,14 @@ class VulkanLayerController {
     }
   }
 
-  void AddRequiredDeviceExtensionNameIfMissing(
-      const VkDeviceCreateInfo* create_info, VkPhysicalDevice physical_device,
-      PFN_vkGetInstanceProcAddr next_get_instance_proc_addr_function, const char* extension_name,
-      std::vector<std::string>* output) {
+  void AddRequiredDeviceExtensionNameIfMissing(const VkDeviceCreateInfo* create_info,
+                                               VkPhysicalDevice physical_device,
+                                               const char* extension_name,
+                                               std::vector<std::string>* output) {
     auto raw_enumerate_device_extension_properties_function =
         // Pass a valid instance, as following the spec. we are not allowed to use nullptr here.
         absl::bit_cast<PFN_vkEnumerateDeviceExtensionProperties>(
-            next_get_instance_proc_addr_function(dispatch_table_.GetInstance(physical_device),
```

Comment:
> For most instance functions, like "vkEnumerateDeviceExtensionProperties"
the Vulkan loader assumes that after the "vkCreateInstance" call
you are not doing another lookup using the pfnNextGetInstanceProcAddr
pointer, but instead create a dispatch table.
If layer A is before our Vulkan layer, we might end up in
getting the function pointer of layer A and fail.

I'm trying to understand this better. So after calling `vkCreateInstance` the loader might return something unexpected from `pfnNextGetInstanceProcAddr`? Why? What does "might" mean in this case? Is it clear why a different function pointer is returned, but only if a layer is loaded before our layer?

Reply:
See [loader_init_instance_core_dispatch_table](https://github.com/KhronosGroup/Vulkan-Loader/blob/16d5d8f25452f207d32d3dec95c0531b8527728e/loader/loader.c#L4724) at the end of loader trampoline, it fills the dispatch table with the top layer function pointer, replacing the icd ones (if the layer defined those function)

Reply:
Thanks!